tools/pre_build.py

```
@@ -0,0 +1,51 @@
+        sys.exit(1)
+    except Exception as e:
+        print(f"✗ Error running {script.name}: {e}")
+        sys.exit(1)
+
+print("\n" + "=" * 60)
+print("✓ All pre-build scripts completed successfully")
+print("=" * 60 + "\n")
+
+
+if __name__ == "__main__":
+    main()
```

Ruff (0.15.6)

[error] Undefined name `main`

(F821)